stl/inc/vector

```diff
@@ -3090,8 +3090,8 @@ _CONSTEXPR20 void _Fill_vbool(_VbIt _First, _VbIt _Last, const _Ty& _Val) {
     }
 }
 
-template <class _VbIt, class _Ty>
-_NODISCARD _CONSTEXPR20 _VbIt _Find_vbool(_VbIt _First, const _VbIt _Last, const _Ty& _Val) {
+template <class _VbIt>
+_NODISCARD _CONSTEXPR20 _VbIt _Find_vbool(_VbIt _First, const _VbIt _Last, const bool _Val) noexcept {

```

Comment:
For the record (no change requested) - I verified that this is correct because the callsites are (as discussed in https://github.com/microsoft/STL/pull/2203#discussion_r709389441 ):
https://github.com/microsoft/STL/blob/88332706f49d3297b200b5605ef174fc47d38d1a/stl/inc/xutility#L5180-L5181
https://github.com/microsoft/STL/blob/88332706f49d3297b200b5605ef174fc47d38d1a/stl/inc/xutility#L5285-L5286
This is important because comparing `true == 1729` is false, so we can't smash `1729` to `bool`.

The `fill` callsites permit any type and I believe that performing a single implicit conversion should be fine according to WG21-N4892 25.7.6 [alg.fill]/2 "*Mandates:* The expression `value` is writable (23.3.1) to the output iterator."
https://github.com/microsoft/STL/blob/88332706f49d3297b200b5605ef174fc47d38d1a/stl/inc/xutility#L4407-L4408
https://github.com/microsoft/STL/blob/88332706f49d3297b200b5605ef174fc47d38d1a/stl/inc/xutility#L4448-L4450